homeassistant/components/fritz/config_flow.py

```diff
@@ -90,7 +90,7 @@ def fritz_tools_init(self) -> str | None:
 
     async def async_check_configured_entry(self) -> ConfigEntry | None:
         """Check if entry is configured."""
-
+        assert self._host
```

Comment:
I don't see a path where it could be `None`? Maybe we don't need to assert but should the type be adjusted?

Reply:
Not sure it will be `None`, but just theoretical `ssdp_location.hostname` can be `str | None`.
https://github.com/home-assistant/core/blob/0006629ca23455eb3d16ad28774660a3e84a7a45/homeassistant/components/fritz/config_flow.py#L125-L126

Additionally, `self._host` is initialized with `None`.
https://github.com/home-assistant/core/blob/0006629ca23455eb3d16ad28774660a3e84a7a45/homeassistant/components/fritz/config_flow.py#L53